tools/lib/gzipTileset.js

```diff
@@ -0,0 +1,110 @@
+ * @param {Boolean} [options.gzip=true] Whether to gzip or gunzip the tileset.
+ * @param {Boolean} [options.tilesOnly=false] Only gzip tiles, does not gzip tileset.json or other files.
+ * @param {Boolean} [options.verbose=false] If true prints out debug messages to the console.
+ */
+function gzipTileset(inputDirectory, outputDirectory, options) {
+    return new Promise(function(resolve, reject) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+        var gzip = defaultValue(options.gzip, true);
+        var tilesOnly = defaultValue(options.tilesOnly, false);
+        var verbose = defaultValue(options.verbose, false);
+
+        if (!defined(inputDirectory)) {
+            reject(new DeveloperError('inputDirectory is required'));
+        }
+        inputDirectory = path.normalize(inputDirectory);
+        outputDirectory = path.normalize(defaultValue(outputDirectory,
+            path.join(path.dirname(inputDirectory), path.basename(inputDirectory) + '-' + (gzip ? 'gzipped' : 'gunzipped'))));
+
+        if (verbose) {
+            console.log('Input directory: ' + inputDirectory);
+            console.log('Output directory: ' + outputDirectory);
+        }
+
+        var files = [];
+        fsExtra.walk(inputDirectory)
+            .on('data', function (item) {
+                if (!item.stats.isDirectory()) {
+                    files.push(path.relative(inputDirectory, item.path));
+                }
+            })
+            .on('end', function () {
+                if (verbose) {
+                    console.log(files.length + ' files found.');
+                    console.log((gzip ? 'Compressing' : 'Uncompressing') + ' files...');
+                }
+                Promise.map(files, function (file) {
+                    var inputFile = path.join(inputDirectory, file);
+                    var outputFile = path.join(outputDirectory, file);
+
+                    if (gzip && tilesOnly && !isTile(inputFile)) {
+                        return fsExtraCopy(inputFile, outputFile);
+                    }
+
+                    return isGzipped(inputFile)
+                        .then(function(fileIsGzipped) {
+                            if (fileIsGzipped === gzip) {
+                                return fsExtraCopy(inputFile, outputFile);
+                            }
+                            return fsExtraEnsureDir(path.dirname(outputFile))
+                                .then(function () {
+                                    var inp = fsExtra.createReadStream(inputFile);
+                                    var out = fsExtra.createWriteStream(outputFile);
+                                    var operation = gzip ? zlib.createGzip() : zlib.createGunzip();
+                                    return streamToPromise(inp.pipe(operation).pipe(out));
+                                });
+
+                        });
+                }, {concurrency: 1024})
```

Comment:
For a cli program, 1024 is fine, but if you want to be conservative, a number like 100 should work fine too and I doubt you would see a performance difference since you are likely to starve the drive you're reading from in both cases.

This function would need a lot of work before I would consider using it in a server environment.  At a quick glance:

1. You wouldn't buffer the entire list of files in memory up front, that would cause problems for huge tilesets
2. Output would never go to disk, instead I would expect a callback to be raised for each tile as it is zipped and let the caller decide where the data gets written.
3. You wouldn't call `console.log` directly, logging/progress would be the of the callback in 1.